File: ruby/pe005.rb
#!/usr/bin/env ruby
# http://projecteuler.net/problem=5

def factor(n)
  factors = Hash.new(0)
  (2..n).each do |divisor|
    while n % divisor == 0
      n /= divisor
      factors[divisor] += 1
    end
    break if n <= 1
  end
  factors
end

factors = Hash.new(0)
(1..20).each do |n|
  factor(n).each do |x, power|
    factors[x] = power if power > factors[x]
  end
end
result = 1
factors.each do |x, power|
  result *= x ** power
end
puts result

File: go/pe005.go
// http://projecteuler.net/problem=5
package main

import (
	"./euler"
	"fmt"
	"math"
)

func main() {
	var factors map[int]float64 = make(map[int]float64)
	for n := 1; n <= 20; n++ {
		for k, v := range euler.Factor(n) {
			factors[k] = math.Max(float64(factors[k]), float64(v))
		}
	}
	result := 1.
	for k, v := range factors {
		result *= math.Pow(float64(k), v)
	}
	fmt.Println(int(result))
}
